### DISCUSSION TITLE
Can't install GCC on Steam Deck Linux

### DISCUSSION THREAD

**Discussion by ThomasEricB** [Homebrew/discussions#4086] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
(127)(deck@steamdeck ~)$ brew config
HOMEBREW_VERSION: 3.6.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: e76c55e18434c870cca1de7aadf9d5218271fe2d
Last commit: 7 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3145eebc9d223953120e4acb2ff29a1e3da27866
Core tap last commit: 4 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit zen
Clang: N/A
Git: 2.38.1 => /bin/git
Curl: 7.85.0 => /bin/curl
Kernel: Linux 5.13.0-valve36-1-neptune x86_64 GNU/Linux
OS: SteamOS Holo (Holo)
Host glibc: 2.36
/usr/bin/gcc: N/A
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: 12.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
(deck@steamdeck ~)$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: No developer tools installed.
Install Clang or run `brew install gcc`.

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/glibc.rb
```

### Description of issue

Hello,

I've been following this tutorial on how to install Homebrew on the Steam Deck (https://gist.github.com/uyjulian/105397c59e95f79f488297bb08c39146) but it won't allow me to install GCC at the end. Here is the issue GCC is having:
```

==> Pouring gcc--12.2.0.x86_64_linux.bottle.2.tar.gz
Warning: The post-install step did not complete successfully
You can try again using:
  brew postinstall gcc
==> Summary
🍺  /home/linuxbrew/.linuxbrew/Cellar/gcc/12.2.0: 1,623 files, 306.6MB
==> Running `brew cleanup gcc`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

Please help

**Comment by SMillerDev** [comment]
What is the output when you run `brew postinstall --debug gcc`?

**Reply by exstrim401** [reply]
https://pastebin.com/rxp1F2TJ

**Reply by SMillerDev** [reply]
What is the output of `/usr/bin/cc -print-file-name=crti.o`?

**Reply by exstrim401** [reply]
> What is the output of `/usr/bin/cc -print-file-name=crti.o`?

No such file or directory

**Comment by retifrav** [comment]
I had the same problem (*everyone did (and does), I reckon*). But it somewhat makes sense, because per [Homebrew requirements](https://docs.brew.sh/Homebrew-on-Linux#requirements) one needs to have `build-essential` package installed in the system, and that missing file (`crti.o`) seems to belong to one of that package components, probably even to GCC itself, so basically in order to install GCC via Homebrew you need to have GCC already available in the system `¯\_(ツ)_/¯`.

There are probably better options, but what I did is I gave up on trying to do that without disabling SteamOS readonly mode, and just disabled it, ran `sudo pacman -S base-devel` and enabled readonly mode back. After that `brew` operation succeeded:

```
$ brew postinstall gcc
==> Postinstalling gcc
==> Creating the GCC specs file: /home/linuxbrew/.linuxbrew/Cellar/gcc/12.2.0/bin/../lib/gcc/current/gcc/x86_64-pc-linux-gnu/12/specs

$ gcc --version
gcc (GCC) 12.2.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Sure, the system's GCC installation (*not Homebrew's*) is likely to get wiped out on the next SteamOS update, but that's fine, as it's not hard to install it back (*if it will be needed again*).

**Comment by gjtiquia** [comment]
yup disabling steamOS readonly mode helps thanks!

here is what i did that works, hope it helps the next person reading this

```bash
# enables dev mode and disables readonly mode
sudo steamos-devmode enable

# installs build-essential
sudo pacman -S base-devel

# re-enable readonly mode
sudo steamos-readonly enable
```

when using `gcc`, i encountered some errors about missing header files
```
fatal error: stdio.h: No such file or directory
```

was able to fix that by running this command (do note that this command takes some time to run)
```
sudo steamos-unminimize --dev
```
